test/unit/range/iterator_test_template.hpp

```diff
@@ -201,8 +193,17 @@ inline void move_forward_pre_test(it_begin_t && it_begin, it_sentinel_t && it_en
     // pre-increment
     auto rng_it = std::ranges::begin(rng);
     auto rng_it_end = std::ranges::end(rng);
-    for (auto it = it_begin; it != it_end && rng_it != rng_it_end; ++it, ++rng_it)
-        expext_eq<test_type>(*it, *rng_it);
+    for (auto it = it_begin; true;)
+    {
+        auto it_copy = ++it;
+        ++rng_it;
+
+        if (!(it != it_end && rng_it != rng_it_end))
```

Comment:
simplify the expression:
```suggestion
        if (it == it_end || rng_it == rng_it_end)
```